python/src/phylanx.cpp

```diff
@@ -1,29 +1,33 @@
 //  Copyright (c) 2017 Hartmut Kaiser
+//  Copyright (c) 2018 R. Tohid
 //
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #include <phylanx/phylanx.hpp>
 
-#include <pybind11/pybind11.h>
 #include <pybind11/operators.h>
+#include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
-#include <hpx/runtime/threads/run_as_hpx_thread.hpp>
 #include <hpx/runtime/components/new.hpp>
+#include <hpx/runtime/threads/run_as_hpx_thread.hpp>
 
 #include <map>
-#include <string>
 #include <sstream>
+#include <string>
 #include <vector>
 
+#include <phylanx/execution_tree/primitives/file_read_csv.hpp>
+#include <phylanx/execution_tree/primitives/slicing_operation.hpp>
```

Comment:
If we decide to expose the primitive components as python objects we need to add all of the related #includes. It might even be a good idea to create a separate file just for those bindings to avoid for this file to become too large.

Reply:
Probably. Is that a blocker?

Reply:
`#including` all needed files is probably a good idea, if for nothing else than readability and robustness. Experience shows that relying on implicit availability of types makes the code base very brittle to unrelated changes (i.e. somebody removing an #include from a different header file might break the compilation of this source).

Separating things into a different source file is probably not a blocker, however as commented on earlier - I would like to see the two (independent) changesets that are currently proposed by this PR to be separated into independent pull requests.